### PR TITLE
ocm: split into core and internal sharing

### PR DESCRIPTION
**Pull request by labkode**
@LovisaLugnegard please add the same methods to the gateway.

@ishank011 that would be the API that is needed to implement both in the grpc gateway (forward) and a new ocmcore provider.

### CODE REVIEW

File: cs3/ocm/core/v1beta1/ocm_core_api.proto [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  // REQUIRED.
+  // Provider specific identifier of the user that wants to share the resource.
+  cs3.identity.user.v1beta1.UserId owner = 5;
+  // REQUIRED.
+  // Consumer specific identifier of the user or group the provider wants to share the resource with.
+  // This is known in advance, for example using the OCM invitation flow.
+  // Please note that the consumer service endpoint is known in advance as well, so this is no part of the request body.
+  // TODO: this field needs to represent either a user or group in the future, not only a user.
+  cs3.identity.user.v1beta1.UserId share_with = 6;
+  Protocol protocol = 7;
```
Comment (ishank011) [inline cs3/ocm/core/v1beta1/ocm_core_api.proto:79]: Please add a comment describing 'protocol'.

**Comment by LovisaLugnegard** [comment]
Looks good to me too!

File: cs3/ocm/core/v1beta1/ocm_core_api.proto (hunk 1 above)
Comment (labkode) [reply cs3/ocm/core/v1beta1/ocm_core_api.proto:79]: I'll add the comment :)

File: cs3/ocm/core/v1beta1/ocm_core_api.proto [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+service OcmCoreAPI {
+  // Creates a new ocm share.
+  rpc CreateOCMCoreShare(CreateOCMCoreShareRequest) returns (CreateOCMCoreShareResponse);
+}
+
+// https://rawgit.com/GEANT/OCM-API/v1/docs.html#null%2Fpaths%2F~1shares%2Fpost
+message CreateOCMCoreShareRequest {
+  // OPTIONAL.
+  // Opaque information.
+  cs3.types.v1beta1.Opaque opaque = 1;
```
Comment (ishank011) [inline cs3/ocm/core/v1beta1/ocm_core_api.proto:60]: We need to add the share_with field as well.

Comment (LovisaLugnegard) [reply cs3/ocm/core/v1beta1/ocm_core_api.proto:60]: Isn't that on line 78?

Comment (ishank011) [reply cs3/ocm/core/v1beta1/ocm_core_api.proto:60]: My bad -_-

Comment (LovisaLugnegard) [reply cs3/ocm/core/v1beta1/ocm_core_api.proto:60]: Np :)

**Comment by labkode** [comment]
@LovisaLugnegard @ishank011 good to merge?

**Pull request merged**